### PR TITLE
IDEA-346847: Fix logic for determining whether to paint left border

### PR DESCRIPTION
**Pull request by sprigogin**

### CODE REVIEW

**Comment by sprigogin** [comment]
Please don't merge yet. Found more issues.

File: platform/platform-impl/src/com/intellij/toolWindow/InternalDecoratorImpl.kt [hunk 1]
```diff
@@ -593,10 +593,11 @@ class InternalDecoratorImpl internal constructor(
         ToolWindowAnchor.LEFT -> !isSplitter || isVerticalSplitter || !isFirstInSplitter
         else -> false
       }
-      val top = if (isSplitter && (anchor == ToolWindowAnchor.RIGHT || anchor == ToolWindowAnchor.LEFT) && windowInfo.isSplit && isVerticalSplitter) -1 else 0
+      val top = if (toolWindow.type != ToolWindowType.FLOATING && (!isVerticalSplitter || isFirstInSplitter) &&
```
Comment (stachenov) [inline platform/platform-impl/src/com/intellij/toolWindow/InternalDecoratorImpl.kt:596]: I don't understand what's going on here yet. The issue is about left/right borders, but these changes only affect the top border.

Comment (sprigogin) [reply platform/platform-impl/src/com/intellij/toolWindow/InternalDecoratorImpl.kt:596]: The change in line 573 affects all borders, but I'm reworking this change to fix remaining issues. Stay tuned.

**Comment by sprigogin** [comment]
Replaced by https://github.com/JetBrains/intellij-community/pull/2686.

**Pull request closed without merging**